File: app/src/main/java/com/loopeer/codereader/event/DownloadProgressEvent.java
package com.loopeer.codereader.event;

public class DownloadProgressEvent {

    public String repoId;
    public long downloadId;
    public float factor;
    public boolean isUnzip;

    public DownloadProgressEvent() {
    }

    public DownloadProgressEvent(long downloadId, float factor) {
        this.downloadId = downloadId;
        this.factor = factor;
    }

    public DownloadProgressEvent(long downloadId, boolean isUnzip) {
        this.downloadId = downloadId;
        this.factor = 1.f;
        this.isUnzip = isUnzip;
    }

    public DownloadProgressEvent(long downloadId, float factor, boolean isUnzip) {
        this.downloadId = downloadId;
        this.factor = factor;
        this.isUnzip = isUnzip;
    }

    public DownloadProgressEvent(String repoId, long downloadId, float factor, boolean isUnzip) {
        this.repoId = repoId;
        this.downloadId = downloadId;
        this.factor = factor;
        this.isUnzip = isUnzip;
    }
}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/event/DownloadProgressEvent.kt
package com.loopeer.codereaderkt.event


class DownloadProgressEvent {

    public lateinit var repoId: String
    public var downloadId: Long = 0
    public var factor: Float = 0.0f
    public var isUnzip: Boolean = false

    constructor(downloadId: Long, isUnzip: Boolean) {
        this.downloadId = downloadId
        this.factor = 1f
        this.isUnzip = isUnzip
    }

    constructor(repoId: String, downloadId: Long, factor: Float, isUnzip: Boolean) {
        this.repoId = repoId
        this.downloadId = downloadId
        this.factor = factor
        this.isUnzip = isUnzip
    }


}
